### PR TITLE
Java class typet methodt

### PR DESCRIPTION
**Pull request by kroening**
- [X] Each commit message has a non-empty body, explaining why the change was made.
- [X] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: jbmc/src/java_bytecode/java_types.h [hunk 1]
```diff
@@ -99,6 +99,102 @@ inline bool can_cast_type<annotated_typet>(const typet &)
...
+
+  const std::vector<irep_idt> throws_exceptions() const
+  {
+    std::vector<irep_idt> exceptions;
+    for(const auto &e : find(ID_exceptions_thrown_list).get_sub())
+      exceptions.push_back(e.id());
+    return exceptions;
+  }
+
+  void add_throws_exceptions(irep_idt exception)
```
Comment (smowton) [inline jbmc/src/java_bytecode/java_types.h:134]: exception, singular

**Review by smowton — APPROVED** [review]

File: jbmc/src/java_bytecode/java_types.h [hunk 2]
```diff
@@ -233,22 +285,16 @@ class java_class_typet:public class_typet
     }
   };
 
-  using componentst = std::vector<componentt>;
+  using methodst = std::vector<methodt>;
 
-  const componentst &components() const
+  const methodst &methods() const
```
Comment (thk123) [inline jbmc/src/java_bytecode/java_types.h:290]: Are there any users for this? It'd be good to include at least some in this PR so doesn't get deleted Perhaps a unit test for checking that a java class when loaded has the methods set?

File: jbmc/src/java_bytecode/java_types.h (hunk 1 above)
Comment (kroening) [reply jbmc/src/java_bytecode/java_types.h:134]: Agreed, but note that this is just moving existing code around. Will do a separate PR.

File: jbmc/src/java_bytecode/java_types.h (hunk 2 above)
Comment (kroening) [reply jbmc/src/java_bytecode/java_types.h:290]: Coming!

**Pull request merged**